lib/theory/src/Theory/Text/Parser/Token.hs

```diff
@@ -235,11 +235,11 @@ naturalSubscript = T.lexeme spthy $ do
 
 -- | A comma separated list of elements.
 commaSep :: Parser a -> Parser [a]
-commaSep = T.commaSep spthy
+commaSep p = option [] (commaSep1 p)
 
 -- | A comma separated non-empty list of elements.
 commaSep1 :: Parser a -> Parser [a]
-commaSep1 = T.commaSep1 spthy
+commaSep1 = (T.commaSep1 <* T.comma) spthy
```

Comment:
for starters, AIUI this should make the comma mandatory (because I forgot `optional` -- and it does not).